java/src/main/java/de/gdata/vaas/VaasConfig.java

```diff
@@ -33,4 +38,9 @@ public VaasConfig(URI tokenEndpoint, URI url) {
         this.url = url;
         this.PullDelayMs = 100;
     }
+
+    public VaasConfig(URI tokenEndpoint, URI url, Duration timeout) {
```

Comment:
Not sure about Java here, but if you set a defaultTimeout in line 28 and override it here anyways, way set it at all? I would not set the timeout in the constructor at all, or provide a second constructor with that argument.